drivers/dma/dma_nxp_4ch_dma.c

```diff
@@ -0,0 +1,469 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/barrier.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/atomic.h>
+
+LOG_MODULE_REGISTER(nxp_4ch_dma, CONFIG_DMA_LOG_LEVEL);
+
+#define DT_DRV_COMPAT	nxp_4ch_dma
+
+struct nxp_dma_chan_data {
+	const struct device *dev;
+	dma_callback_t cb;
+	void *user_data;
+	uint32_t width;
+	uint8_t src_inc;
+	uint8_t dst_inc;
+	enum dma_channel_direction dir;
+	bool busy;
+	uint16_t dmamux_source;
+};
+
+struct nxp_dma_data {
+	struct dma_context ctx;
+	struct nxp_dma_chan_data *chan;
+};
+
+struct nxp_dma_config {
+	DMA_Type *dma_base;
+	DMAMUX_Type *dmamux_base;
+	uint8_t num_channels;
+	uint8_t dma_requests;
+	void (*irq_config_func)(const struct device *dev);
+#if IS_ENABLED(CONFIG_CLOCK_CONTROL)
```

Comment:
Will this driver work without clock control? I'm highly skeptical from what I'm seeing here. If you *need* clock control, then use Kconfig to ensure its selected when this driver is being built rather than ifdef'ing. If you can work without it I'd be curious to see how, are the clocks setup at soc or board init time?